### PR TITLE
Feature/dp widget

### PR DESCRIPTION
**Pull request by al3xknutsen**
A dynamic positioning widget now shows information about the ROVs position, the target position, and the distances between the two:
image

The target will be displayed in the minimap:
image

Also fixed a bug where negative coordinates would cause the ROV arrow to be rendered incorrectly if the ROV was outside the minimap :)

### CODE REVIEW

**Comment by hsyver** [comment]
Looks like east and north has been switched on the minimap.

oops2

And the netfollowing widget is placed to the side when I have the video window not fullscreen.

oops

Other than that, this looks very nice 👏

**Comment by al3xknutsen** [comment]
@hsyver Thank you, should be fixed now :)

File: src/VideoComponents/ModeWidget.jsx [hunk 1]
```diff
@@ -20,6 +21,13 @@ class ModeWidget extends Component {
   constructor(props) {
     super(props);
 
+    this.state = {
+      fontSizeMode: 14,
+      fontSizeNFAvail: 14,
+      sizeImgNFAvail: 25,
+      hey: 'hey',
```
Comment (Trebua) [inline src/VideoComponents/ModeWidget.jsx]: remove hey

**Comment by al3xknutsen** [comment]
@Trebua Thank you :) I have noticed the little jump, to make this work properly I think one would have to use React refs to access the DOM elements when scaling. I didn't get that to work unfortunately, and resorted to a solution with React state and an onLoad function. If anyone knows React refs better than I do you're welcome to give it a go :P

To test the target on the minimap one would have to turn on DP mode first. This seems to always be disabled right now, so one would have to manually enable it in code.

**Pull request merged**